db/model/Gdoc/Gdoc.ts

```diff
@@ -33,39 +34,57 @@ export class Gdoc extends BaseEntity implements OwidArticleType {
         }
         this.content = {}
     }
-    static cachedGoogleAuth?: Auth.GoogleAuth
+    static cachedGoogleReadonlyAuth?: Auth.GoogleAuth
+    static cachedGoogleReadWriteAuth?: Auth.GoogleAuth
     static cachedGoogleClient?: docs_v1.Docs
 
-    static getGoogleAuth(): Auth.GoogleAuth {
-        if (!Gdoc.cachedGoogleAuth) {
-            Gdoc.cachedGoogleAuth = new google.auth.GoogleAuth({
+    static getGoogleReadWriteAuth(): Auth.GoogleAuth {
+        if (!Gdoc.cachedGoogleReadWriteAuth) {
+            Gdoc.cachedGoogleReadWriteAuth = new google.auth.GoogleAuth({
                 credentials: {
                     type: "service_account",
                     private_key: GDOCS_PRIVATE_KEY.split("\\n").join("\n"),
                     client_email: GDOCS_CLIENT_EMAIL,
                     client_id: GDOCS_CLIENT_ID,
                 },
-                // Scopes can be specified either as an array or as a single, space-delimited string.
-                scopes: ["https://www.googleapis.com/auth/documents.readonly"],
+                scopes: [
+                    "https://www.googleapis.com/auth/documents",
+                    "https://www.googleapis.com/auth/drive.file",
+                ],
             })
         }
-        return Gdoc.cachedGoogleAuth
+        return Gdoc.cachedGoogleReadWriteAuth
     }
 
-    static getGoogleClient(): docs_v1.Docs {
-        if (!Gdoc.cachedGoogleClient) {
-            Gdoc.cachedGoogleClient = google.docs({
-                version: "v1",
-                auth: Gdoc.cachedGoogleAuth,
+    static getGoogleReadonlyAuth(): Auth.GoogleAuth {
+        if (!Gdoc.cachedGoogleReadonlyAuth) {
+            Gdoc.cachedGoogleReadonlyAuth = new google.auth.GoogleAuth({
+                credentials: {
+                    type: "service_account",
+                    private_key: GDOCS_PRIVATE_KEY.split("\\n").join("\n"),
+                    client_email: GDOCS_CLIENT_EMAIL,
+                    client_id: GDOCS_CLIENT_ID,
+                },
+                scopes: ["https://www.googleapis.com/auth/documents.readonly"],
             })
         }
-        return Gdoc.cachedGoogleClient
+        return Gdoc.cachedGoogleReadonlyAuth
+    }
+
+    async createGdocArticleFromArchieMl(
+        article: OwidArticleContent
+    ): Promise<void> {
+        this.id = await createGdocAndInsertOwidArticleContent(article)
+
+        this.content = article
+
+        await this.save()
     }
```

Comment:
so this will be used when closing the loop in the follow-up PR to insert the gdoc in the `posts_gdocs` table (TODO comment in `createGdoc` route)?

Reply:
actually, I'm confused about this. If we were to call it in addition to the `createGdoc` route, we would end up creating a duplicate doc on gdrive. Is this for the future bulk import maybe?

Reply:
Yeah I removed this one again and will introduce it again when I come to doing the full circle